Sources/Commands/Describe.swift

```diff
@@ -10,85 +10,354 @@
-        stream <<< Format.asRepeating(string: " ", count: indent)
-            <<< "Name: " <<< name <<< "\n"
-        stream <<< Format.asRepeating(string: " ", count: indent)
-            <<< "C99name: " <<< c99name <<< "\n"
-        stream <<< Format.asRepeating(string: " ", count: indent)
-            <<< "Type: " <<< type.rawValue <<< "\n"
-        stream <<< Format.asRepeating(string: " ", count: indent)
-            <<< "Module type: " <<< String(describing: Swift.type(of: self)) <<< "\n"
-        stream <<< Format.asRepeating(string: " ", count: indent)
-            <<< "Path: " <<< sources.root.pathString <<< "\n"
-        stream <<< Format.asRepeating(string: " ", count: indent)
-            <<< "Sources: " <<< sources.relativePaths.map({ $0.pathString }).joined(separator: ", ") <<< "\n"
+    /// Represents a product for the sole purpose of generating a description.
+    struct DescribedProduct: Encodable {
+        let name: String
+        let type: ProductType
+        let targets: [String]
+
+        init(from product: Product, in package: Package) {
+            self.name = product.name
+            self.type = product.type
+            self.targets = product.targets.map { $0.name }
+        }
     }
 
-    public func toJSON() -> JSON {
-        return .init([
-            "name": name,
-            "c99name": c99name,
-            "type": type,
-            "module_type": String(describing: Swift.type(of: self)),
-            "path": sources.root,
-            "sources": sources
-        ])
+    /// Represents a target for the sole purpose of generating a description.
+    struct DescribedTarget: Encodable {
+        let name: String
+        let type: String
+        let c99name: String?
+        let moduleType: String?
+        let path: String
+        let sources: [String]
+        let resources: [PackageModel.Resource]?
+        
+        init(from target: Target, in package: Package) {
+            self.name = target.name
+            self.type = target.type.rawValue
+            self.c99name = target.c99name
+            self.moduleType = String(describing: Swift.type(of: target))
+            self.path = target.sources.root.relative(to: package.path).pathString
+            self.sources = target.sources.relativePaths.map{ $0.pathString }
+            self.resources = target.resources.isEmpty ? nil : target.resources
+        }
     }
 }
 
-extension Sources: JSONSerializable {
-    public func toJSON() -> JSON {
-        return .array(relativePaths.map({ .string($0.pathString) }))
+
+public struct PlainTextEncoder {
+    
+    /// The formatting of the output plain-text data.
+    public struct FormattingOptions: OptionSet {
+        public let rawValue: UInt
+
+        public init(rawValue: UInt) {
+            self.rawValue = rawValue
+        }
+
+        /// Produce plain-text format with indented output.
+        public static let prettyPrinted = FormattingOptions(rawValue: 1 << 0)
```

Comment:
Not yet used:  currently the encoder is always pretty-printing (that's the point of plain text).  Might add other options here, or will get rid of the formatting options.